sql_generators/urlbar_events/templates/desktop_query.sql

```diff
@@ -1,190 +1,110 @@
+CREATE TEMP FUNCTION enumerated_array(results ARRAY<STRING>, _groups ARRAY<STRING>) AS (
+  ARRAY(
+    SELECT
+      STRUCT(
+        off + 1 AS position,
+        r AS result_type,
+        `mozfun.norm.result_type_to_product_name`(r) AS product_result_type,
+        _groups[SAFE_OFFSET(off)] AS result_group
+      )
+    FROM
+      UNNEST(results) AS r
+      WITH OFFSET off
+  )
+);
+
+CREATE TEMP FUNCTION get_event_action(event_name STRING, engagement_type STRING) AS (
+  CASE
+    WHEN event_name = 'engagement'
+      AND (engagement_type IN ("click", "drop_go", "enter", "go_button", "paste_go"))
+      THEN 'engaged'
+    WHEN event_name = 'abandonment'
+      THEN 'abandoned'
+    WHEN event_name = 'engagement'
+      AND (engagement_type NOT IN ("click", "drop_go", "enter", "go_button", "paste_go"))
+      THEN "annoyance"
+    ELSE NULL
+  END
+);
+
+CREATE TEMP FUNCTION get_is_terminal(selected_result STRING, engagement_type STRING) AS (
+  --events where the urlbar dropdown menu remains open (i.e., the urlbar session did not end)
+  COALESCE(
+    NOT (selected_result = 'tab_to_search' AND engagement_type IN ('click', 'enter', 'go_button'))
+    AND NOT (
+      selected_result = 'tip_dismissal_acknowledgement'
+      AND engagement_type IN ('click', 'enter')
+    )
+    AND NOT (
+      engagement_type IN (
+        'dismiss',
+        'inaccurate_location',
+        'not_interested',
+        'not_relevant',
+        'show_less_frequently'
+      )
+    ),
+    TRUE
+  )
+);
+
 WITH events_unnested AS (
   SELECT
     DATE(submission_timestamp) AS submission_date,
+    client_info.client_id AS glean_client_id,
+    metrics.uuid.legacy_telemetry_client_id AS legacy_telemetry_client_id,
     sample_id,
-    client_info.client_id,
+    name AS event_name,
```

Comment:
Can we add the `event` alias for fields that are pulled from there? This one is one, right (`event.name AS event_name`)?